pkg/logging/logging.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2021 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package fybriklogging
+
+import (
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+)
+
+// LogType indicates the types of log entries supported.  Logging packages typically provide separate function
+// calls for the different log types.
+type LogType string
+
+const (
+	ERROR LogType = "error" // Errors encountered in the code
+	INFO  LogType = "info"  // Info logs are things you want to tell the user or administrator which are not errors.
+)
+
+// VerbosityLevel allows one to decide how "chatty" you want your logs to be.
+type VerbosityLevel int
+
+const (
+	MANDATORY VerbosityLevel = 0
+	DEBUG     VerbosityLevel = 1
+	TRACE     VerbosityLevel = 2
+)
+
+// LogEntry defines the information for a single entry stored in the system logs.
+// It is assumed that the timestamp is added automatically and thus not included in the log entry structure.
+// Verbosity and log type are added by the log library via which the entries are sent to stdout or stderr.
+type LogEntry struct {
+	// Caller indicates the cluster, component and function that generated the message in the format "cluster/component/function"
+	// Use the buildCallerPath function to construct this.
+	// +required
+	Caller string `json:"caller"`
+
+	// Message being logged.
+	// +required
+	Message string `json:"message"`
+
+	// Unique id of the kubernetes fybrikapplication (Needs to be unique over time in order to support history and not just current workloads)
+	// +required
+	FybrikAppGUID string `json:"fybrikAppGuid"`
+
+	// ForUser should be true if the message should be shared with the end user in the FybrikApplication status.
+	// Assumed false if not indicated.
+	// +optional
+	ForUser bool `json:"forUser,omitempty"`
+
+	// ForArchive should be true if the message should be archived long term.
+	// For example, contains full contents of FybrikApplication and its status and should be stored for auditing purposes.
+	// Assumed false if not indicated
+	// +optional
+	ForArchive bool `json:"forArchive,omitempty"`
```

Comment:
I suggest to rename to Archive.

Reply:
Shouldn't ARCHIVE be a logType like INFO?

Reply:
I don't think it is fybrik developer's decision to archive log or not. Probably we should flag or log messages or create a separate log target for auditing. The retention policy is out of our scope.

Reply:
I have to admit that I debated the approach to take here as well.  There are 3 options: (a) have a flag in the log entry structure as I've done here, (b) have a verbosity level specific to archive, or (c) have a separate log for auditing.  

All 3 options require the developer to indicate that it's something that needs to be archived.  He/she of course does not deal with retention policies.  That's out of scope for fybrik, and is handled by those deploying log processing tools.

I did not choose (c) because I think it's preferable that fybrik continue to write (via go-logr) only to std_err and std_out and let the log aggregator filter and route entries as desired rather than putting that logic in fybrik.  

I did not choose (b) because I thought we might want to have the flexibility to combine an archive field and verbosity, meaning that in essence you can also change the verbosity of what gets archived if desired.

@shlomitk1 I'm fine with changing to Archive

Reply:
@simanadler I agree with the chosen option (a), but my point that it is not fybrik responsibility to decide what should be archived or not, therefore I just suggest using a flag "Audit" or something similar. 
Somebody else will decide if he wants to archive it or not.

Reply:
Changed ForArchive to Audit